app/models/blended_search.rb

```diff
@@ -44,40 +44,46 @@ def first_page?
   protected
 
   def handle_response(response)
-    if response
-      @total = response.total
-      post_processor = ResultsWithBodyAndDescriptionPostProcessor.new(response.results)
-      post_processor.post_process_results
-      @results = paginate(response.results)
-      @normalized_results = post_processor.normalized_results
-      @startrecord = ((@page - 1) * @per_page) + 1
-      @endrecord = @startrecord + @results.size - 1
-      assign_spelling_suggestion_if_eligible(response.suggestion.text) if response.suggestion.present?
-    end
+    return unless response
+
+    @total = response.total
+    post_processor = ResultsWithBodyAndDescriptionPostProcessor.new(response.results)
+    post_processor.post_process_results
+    @results = paginate(response.results)
+    process_data_for_redesign(post_processor)
```

Comment:
Unless you have a strong preference for this, I mildly prefer:

```suggestion
    @normalized_results = process_data_for_redesign(post_processor)
```

and removing the `@normalized_results =` from the method itself.  Makes it easier to glance down `handle_response` and see all the instance variables.  

But that's just a minor quibble.  If you do like the idea of updating this, do it also in `i14y_search` and `web_search`.

Reply:
I like this idea and will use it. Thanks for the idea!